.github/actions/detect-env/action.yml

```diff
@@ -33,7 +37,7 @@ runs:
     - name: Detect framework
       shell: bash
       run: |
-        if ${{ inputs.framework || false }}; then
+        if ${{ inputs.framework != '' }}; then
```

Comment:
This won't work since it's bash.
if true and if false works, but this won't.
Is there a reason to change this?

Reply:
@danipavic any feedback here? You did that change, right?

Reply:
When testing out the flow, `inputs.framework || false` syntax did not work, and the command on the next line was never executed. I noticed this when adding the `dist_path` input. I discussed this with Maks and he also concluded that more or less everyone is using this approach to check if an input is present. If there is a better syntax/way to achieve the same thing, go for it.

Reply:
Hmm you are right, my bad. The whole variable will return false if it isn't defined. Sorry.

Reply:
@danipavic 
Sorry, checked again. The `inputs.framework != ''` won't work because it has a default defined as `false` so it will never be `''`. Maybe it would be best to leave it as it was.

Reply:
Brokenships deploy was true because `inputs.framework != ''` returned true (it was set to `false` value by default):
```
Run if true; then
  if true; then
    echo "ssr=false" >> $GITHUB_ENV
  elif [ -f next.config.js ]; then
    echo "ssr=true" >> $GITHUB_ENV
  else
    echo "ssr=false" >> $GITHUB_ENV
  fi
```
@DarkoKukovec That explains all the missing files.

Reply:
My issue was that when input was provided, it still wouldn't resolve the condition as `true`. Eg. I provided the framework `svelte` and I expected that `svelte` is set as the variable, but that is not what happened, `angular` was set instead.
Workflow:
```
with:
      environment: 'staging'
      framework: 'svelte'
      ci_steps: 'lint test deploy'
      deploy_host: 'guess-who-fe-staging.byinfinum.co'
      deploy_user: 'guess_who_staging'
      deploy_port: 229
```

Detect env:
```
if svelte; then
    echo "framework=svelte" >> $GITHUB_ENV
  elif [ -f angular.json ]; then
    echo "framework=angular" >> $GITHUB_ENV
  elif [ -f next.config.js ]; then
    # This is fine for now since we only support standalone builds and this needs to be set in the next config
    echo "framework=next" >> $GITHUB_ENV
  else
    echo "framework=react" >> $GITHUB_ENV
  fi
```

Final output:
```
Framework: angular
SSR: false
Package manager: npm
New Relic: false
dist_path: ./dist/*
artifact_path: ./dist/*
```

Reply:
Ahh, I apologize, I meant `ssr`. I don't know why I wrote framework. That variable has a default as `false` in pipeline.yml and is sent to detect-env. `if ${{ inputs.ssr != '' }}; then` returns `if true`, when it should be `false`.
We should set it to `if ${{ inputs.ssr != 'false' }}; then`.

Reply:
That makes sense, perhaps we don't even need the `default` value anymore?